packages/visualizer/src/store/summarizerStore.ts

```diff
@@ -0,0 +1,107 @@
+import { create } from 'zustand';
+import { immer } from 'zustand/middleware/immer';
+
+export enum SummarizerState {
+  CHECKING_AVAILABILITY,
+  UNAVAILABLE,
+  DOWNLOADING,
+  READY,
+}
+
+type SummarizerStoreState = {
+  summarizerState: SummarizerState;
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  tldrSummarizer: any | null;
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  keyPointsSummarizer: any | null;
+  summarizerDownloadProgress: number;
+};
+
+type SummarizerStoreActions = {
+  ensureInitialized: () => void;
+};
+
+export type SummarizerStore = SummarizerStoreState & SummarizerStoreActions;
+
+export const useSummarizerStore = create<SummarizerStore>()(
+  immer((set) => ({
+    ensureInitialized: async () => {
+      // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+      // @ts-expect-error -- missing types for Summarizer API
+      const availability = await Summarizer.availability();
```

Comment:
For other browsers than Chrome, that line might fail if Summarizer is undefined.

Would be great to add "typeof Summarizer !== 'undefined'" check before using the API